rabbit-escape-render/src/rabbitescape/render/gameloop/GeneralPhysics.java

```diff
@@ -104,6 +124,25 @@ public boolean gameRunning()
         return ( world.completionState() == World.CompletionState.RUNNING );
     }
 
+    /**
+     * Take actions for demo mode, eg drop tokens.
+     */
+    private void doInterpreterActions()
+    {
+        SolutionTimeStep stp = solutionInterpreter.next( CompletionState.RUNNING );
+        for ( TimeStepAction action: stp.actions )
+        {
+            if ( action instanceof SelectAction )
+            {
+                actionTypeSwitch.caseSelectAction( (SelectAction)action );
+            }
+            else if ( action instanceof PlaceTokenAction )
+            {
+                actionTypeSwitch.casePlaceTokenAction( (PlaceTokenAction)action );
+            }
```

Comment:
Not sure reusing TimeStepActionTypeSwitch here is very clear? I can live with it if you defend it but I was expecting these type switch classes to only be implemented inline.